app/src/main/java/com/studhub/app/MainViewModel.kt

```
@@ -0,0 +1,32 @@
+package com.studhub.app
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.studhub.app.core.utils.ApiResponse
+import com.studhub.app.data.network.NetworkStatus
+import com.studhub.app.domain.usecase.conversation.FlushPendingMessages
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class MainViewModel @Inject constructor(
+    private val networkStatus: NetworkStatus,
+    private val flushPendingMessages: FlushPendingMessages
+): ViewModel() {
+    fun getNetworkState() = networkStatus.connectivityStatus().stateIn(
+        viewModelScope,
+        SharingStarted.WhileSubscribed(),
+        networkStatus.isConnected
+    )
```

The code is good, some few improvements here can done: -Try to add some comments, it's always useful for the comprehension of the others. -Lastly but not less important, make sure all potential errors are handled, for examples after the execution of **flushPendingMessages()**.

---

`flushPendingMessages` can fail silently as if it fails, this is most certainly because of connectivity issues. Anyway buffered messages won't be wiped on failure so `flushPendingMessages` will just be called again when the connection is back